cli/src/cli_util.rs

```diff
@@ -2471,7 +2471,7 @@ impl CliRunner {
     fn run_internal(
         self,
         ui: &mut Ui,
-        mut layered_configs: LayeredConfigs,
+        layered_configs: &mut LayeredConfigs,
```

Comment:
This `&mut` and `CommandHelper<'c>` looks scary.

Perhaps, `ClapCliError(..)` can be extended to `{ err, hint }`, and attach the hint in `run_internal()`?

```rust
        let (matches, args) = parse_args(
            ui,
            &self.app,
            &self.tracing_subscription,
            &string_args,
            layered_configs,
        ).map_err(HERE)?;
```